experimental/dds/tree2/src/core/rebase/utils.ts

```diff
@@ -128,7 +132,7 @@ export function rebaseBranch<TChange>(
 	targetHead: GraphCommit<TChange>,
 ): [
 	newSourceHead: GraphCommit<TChange>,
-	sourceChange: TChange | undefined,
+	sourceChange: () => TChange | undefined,
```

Comment:
One thought I had is that you could change the returned tuple to be an object instead. Then, `sourceChange` could just be a getter, so you wouldn't need to type it as a function `() =>` everywhere, and it would be slick.

The downside is that it's less obvious that it's computing expensive work. That's only a problem if we are scared of people using it accidentally (probably not).

Reply:
Or, [you can keep it as a tuple](https://www.typescriptlang.org/play?#code/C4TwDgpgBAKlC8UDaA7ArgWwEYQE4BooBnYXASxQHNCsB7WgGwgEMUBdAbgFgAoXgMzQoAxsDK0UUfgAoAlAC5YUAN68o6qLgjA0uSap4ajUAEQAGE4oAsAJnxrj6yttMBGE3JWbtuySYAWEAwMtCZQAL72ho6mNpZQpGgQDurhUMxEsNw84by8whIkCQhSctkFKESMEAB0IZTSwEhmbLLlhdV1tA1Nrq3tlZ31jUg2-UA)... *evil grin*